### PR TITLE
Add Missing Products Class

### LINKED ISSUE #62: Missing data in CurrentVolatilStockResponse._missing_products
Hi!

First of all many thanks for pygrocy, I have been using it for a while and it helped me a lot!

I think I got a parse error when I try to get the missing products using `grocy.missing_products()`

If I check the JSON I See:

```
"missing_products": [
        {
            "id": "7",
            "name": "XXXX",
            "amount_missing": "1",
            "is_partly_in_stock": "1"
        },
        {
            "id": "8",
            "name": "YYYY",
            "amount_missing": "1",
            "is_partly_in_stock": "1"
        },
        {
            "id": "75",
            "name": "ZZZZ",
            "amount_missing": "1",
            "is_partly_in_stock": "1"
        }
    ]
```

But diving in [CurrentStockResponse](https://github.com/SebRut/pygrocy/blob/47dbb07f93403500f133a13012264aa1439e68db/pygrocy/grocy_api_client.py#L177) is expecting antother property names: 

```
self._product_id = parse_int(parsed_json.get('product_id'))
self._amount = parse_float(parsed_json.get('amount'))
self._best_before_date = parse_date(parsed_json.get('best_before_date'))
self._amount_opened = parse_float(parsed_json.get('amount_opened'))
```

Do we might need a new class to represent missing products?

If so, I can try to open a PR with that change.

### PR DESCRIPTION
**Pull request by GerardRibas**
## Description

Added new class for missing products.
Changed Products constructor to pass in generic object to be parsed.
Added unit test

**Related issue (if applicable):** fixes #62 

## Checklist
  - [x] The code change is tested and works locally.
  - [x] tests pass. **Your PR won't be merged unless tests pass**
  - [x] There is no commented out code in this PR

### CODE REVIEW

File: pygrocy/grocy.py [hunk 1]
```diff
@@ -11,12 +11,21 @@
...
+            self._best_before_date = response.best_before_date
+            self._amount_missing = None
+            self._is_partly_in_stock = None
+        elif type(response) is MissingProductResponse:
+            self._product_id = response.product_id
+            self._name = response.name
+            self._available_amount = None
+            self._best_before_date = None
+            self._amount_missing = response.amount_missing
+            self._is_partly_in_stock = response.is_partly_in_stock
```
Comment (SebRut) [inline pygrocy/grocy.py]: `is`suggests that this is respresenting a boolean. Could you investigate that a abit?

Comment (GerardRibas) [reply pygrocy/grocy.py]: I thought the same, but Grocy documentation is set as an Integer:

```
missing_products : [ 
   id | integer
   name | string
   amount_missing | number
   is_partly_in_stock | integer ]
```

Comment (GerardRibas) [reply pygrocy/grocy.py]: In [0038.sql](https://github.com/grocy/grocy/blob/b81316bd60b3ff96a0bd5add73bbf0a1fbf8dece/migrations/0038.sql) it seems to be a 0 - 1 integer so I think it's safe to change it to boolean. I'll update the PR.

**Comment by SebRut** [comment]
Seems like Travis detected abuse again, I contacted them -.-

**Comment by SebRut** [comment]
Should be resolved, but I guess you need to add another commit, maybe with results from CodeFactor or after running isort over changed files?

**Comment by GerardRibas** [comment]
Seems that I don't have privileges to run the CI pipeline? I see in Travis `Abuse detected` again :(
https://travis-ci.com/SebRut/pygrocy/requests

**Comment by SebRut** [comment]
Nah, it's just their weird abuse detection system. I'll contact them again:-|

**Comment by SebRut** [comment]
They should have unblocked stuff again, if that's not working maybe you could contact them via email, they usually respond in a day.

**Comment by SebRut** [comment]
@GerardRibas it seems like their system is flagging you, could you contact Travis CI per mail please?

**Comment by GerardRibas** [comment]
@SebRut seems that I'm unblocked now! 😄 

Is there a way to trigger CI pipeline without a new commit?

**Comment by SebRut** [comment]
Nice :) I guess you have to change something again, maybe write out "Dev" to "Development" in the README file?

**Pull request merged**